src/legacy/core_plugins/vis_type_vislib/public/vislib/helpers/point_series/_init_x_axis.ts

```diff
@@ -19,31 +19,31 @@
 
 import { uniq } from 'lodash';
 import moment from 'moment';
-import { Chart, FakeParams, DateHistogramParams, HistogramParams } from './point_series';
+import { Chart } from './point_series';
 import { Table } from '../../types';
 
 export function initXAxis(chart: Chart, table: Table) {
   const { format, title, params, accessor } = chart.aspects.x[0];
 
   chart.xAxisOrderedValues =
-    accessor === -1
-      ? [(params as FakeParams).defaultValue]
+    accessor === -1 && 'defaultValue' in params
+      ? [params.defaultValue]
       : uniq(table.rows.map(r => r[accessor]));
   chart.xAxisFormat = format;
   chart.xAxisLabel = title;
 
-  const { interval, date } = params as DateHistogramParams;
-  if (interval) {
-    if (date) {
-      const { intervalESUnit, intervalESValue } = params as DateHistogramParams;
+  if ('interval' in params) {
```

Comment:
This is simpler than casting :smile:.

Reply:
Most important it's safer because Typescript infers the correct type on its own 🙂